### ISSUE TITLE
Provide support/integration for BlueMail decrypt call to "APG"

### ISSUE THREAD

**Issue by avalanchas** [open-keychain/open-keychain#2258]
## Expected Behavior
The E-Mail App [Bluemail](https://bluemail.me/) provides support for encryption and decryption of E-Mail directly in-App. However, I suspect this is done with a direct Intent call to the fully qualified package of APG itself (which is not supported anymore and of which this App here is a fork). More on this below. The expected behaviour is the one that one can witness when you install an old .apk of APG. The call comes back from APG and the encrypted email is translated directly inside Bluemail

## Current Behavior
The process described above doesn't work with "only" OKC installed and there is only an error message telling the user that the request can't be handled inside Bluemail. Currently, the only solution for this is copying the encrypted message, opening OKC and then lettings OKC translate the String in the clipboard

## Possible Solution
To fix this one would need to look into the intent handling of APG, OKC and Bluemail. Then OKC would have to register for the correct intent and provide the functionality (probably simply a string return value?) that Bluemail expects.
I am already in contact with the Bluemail support team but they are reluctant to give technical details. Some analysis of the logcat will be required

## Steps to Reproduce (for bugs)

1. Install [Bluemail App](https://play.google.com/store/apps/details?id=me.bluemail.mail)
2. Set up an email account
3. Send yourself a PGP encrypted email
4. Open the email and press the "decrypt" button on the top

## Context
This feature would allow users of Bluemail to have deepy integrated encryption usage and improve the UX of OKC at the same time. There is a knowledge base article [here](https://bluemail.help/send-receive-encrypted-messages/) that simply refers to APG, but there seems to be at least some interest

Now, I don't know whether this has been tried before or if it's even possible. This is my first time trying to contribute to any Github repo so forgive me if I overstep or anything. I could not find any hint that this has been tried/requested before by searching in the repo and the issues here so I hope I'm not creating duplicates. I also do not know if my request is even in the spirit of OKC, the Intents, the OpenPGP API or whatever we'll have to use to implement this. I'm thankful for any advice as I am starting now to look at the logcat and the intents that APG picks up on my phone

## Your Environment
* Android Version: 7.1.2 (but should be reproducible on any that supports Bluemail and OKC)
* Device Model: Moto X Play
* OpenKeychain Version: 4.8.1
* From Google Play or F-Droid?: Play

**Comment by philipwhiuk** [comment]
Why can't blue mail just implement the API like everyone else.

Frankly the fact they still look for APG suggests you really don't want to use the app.

**Comment by avalanchas** [comment]
Yup, that is a thought that had crossed my mind as well, it's quite weird, I agree. Not only for this project, it's also bad Android practice in general to look for a specific package instead of broadcasting an Intent type in general

So... should I take your comment as a hint to not implement this in OKC, to not make a pull request?
Cause I basically figured it our yesterday evening and I now know the Intent call that they make. I created this small 10-liner demo App to make a Demo:
[Click here for Gist](https://gist.github.com/avalanchas/045e5e4434b75d2ce43f6a49970dbd72)
And I could implement it now. 

I kinda like everything else about BlueMail. Alternatively, could anybody point to an Email App that has this feature (in-app, click a button, have the text decrypted right there on the fly)?

**Comment by philipwhiuk** [comment]
Have you tried K-9 Mail?

As for whether a patch would be accepted, I'm not sure - I'm not a project member.

**Comment by nuntius35** [comment]
According to this [blog](https://www.kuketz-blog.de/blue-mail-android-mail-app-versendet-login-passwort-zum-e-mail-konto/), BlueMail sends your password in clear text to their servers. If this is true, I don't think that anyone should use this app.
